sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerUIEventListener.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.thriftserver.ui
+
+import org.apache.hive.service.cli.SessionHandle
+import org.apache.hive.service.cli.session.HiveSession
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.scheduler.{SparkListenerJobStart, SparkListener}
+
+import scala.collection.mutable.HashMap
+
+trait ThriftServerEventListener {
+
+  def onConnected(session: HiveSession) { }
+  def onDisconnected(session: HiveSession) { }
+
+  def onStart(id: String, session: HiveSession, statement: String) { }
+  def onParse(id: String, executePlan: String, groupId: String) { }
+  def onError(id: String, errorMessage: String, errorTrace: String) { }
+  def onFinish(id: String) { }
+}
+
+class SessionInfo(val session: HiveSession, val startTimestamp: Long) {
+  val sessionID = session.getSessionHandle.getSessionId.toString
+  var finishTimestamp = 0L
+  var totalExecute = 0
+
+  def totalTime = {
+    if (finishTimestamp == 0L) {
+      System.currentTimeMillis() - startTimestamp
+    } else {
+      finishTimestamp - startTimestamp
+    }
+  }
+}
+
+object ExecutionState extends Enumeration {
+  val STARTED, COMPILED, FAILED, FINISHED = Value
+  type ExecutionState = Value
+}
+
+class ExecutionInfo(val statement: String, val session: HiveSession, val startTimestamp: Long) {
+  var finishTimestamp = 0L
+  var executePlan = ""
+  var detail = ""
+  var state: ExecutionState.Value = ExecutionState.STARTED
+  var groupId = ""
```

Comment:
@scwf @liancheng , I noticed that the `groupId` in `SparkExecuteStatementOperation` had been removed in  #3718. What's this `groupId` used for?